hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -646,33 +659,45 @@ private void applyDelta(Blob blob, HollowReadStateEngine stateEngine) throws IOE
     }
 
     private void validate(HollowProducer.ReadState readState) {
-        ProducerStatus.Builder status = listeners.fireValidationStart(readState);
-        try {
-            List<Throwable> validationFailures = new ArrayList<Throwable>();
-            
-            for(Validator validator : validators) {
-                try {
-                    validator.validate(readState);
-                } catch(Throwable th) {
-                    validationFailures.add(th);
-                }
-            }
-            
-            if(!validationFailures.isEmpty()) {
-                ValidationException ex = new ValidationException("Validation Failed", validationFailures.get(0));
-                ex.setIndividualFailures(validationFailures);
-                throw ex;
-            }
-            
-            status.success();
-        } catch (Throwable th) {
-            status.fail(th);
-            throw th;
-        } finally {
-            listeners.fireValidationComplete(status);
-        }
+    	com.netflix.hollow.api.producer.HollowProducerListener.ProducerStatus.Builder psb = listeners.fireValidationStart(readState);
+    	List<Throwable> exceptions = new ArrayList<>();
+    	AllValidationStatusBuilder valStatus = AllValidationStatus.builder(readState.getVersion());
+    	
+    	try {
+    		for(Validator validator: validators) {
+    			Throwable throwable = null;
+    			try {
+    				validator.validate(readState);
+	    		} catch (Throwable th) {
+	    			throwable = th;
+	    			exceptions.add(th);
+	    		}
+    			valStatus.addSingelValidationStatus(getValidationStatus(readState, validator, throwable));
+    		}
+	    	
+	    	if(!exceptions.isEmpty()) {
```

Comment:
wonder if this logic could be encapsulated a little bit different.  For example,

- have valStatus, keep track of all exceptions
- it will also be responsible for tracking fail/success automatically while processing individual validator results

Reply:
Will give it another try in next PR.

Reply:
(saw your comment after merging)